Source/WebCore/platform/graphics/gstreamer/VideoTrackPrivateGStreamer.cpp

```diff
@@ -59,66 +58,71 @@ VideoTrackPrivateGStreamer::VideoTrackPrivateGStreamer(WeakPtr<MediaPlayerPrivat
-        gst_stream_set_stream_flags(m_stream.get(), static_cast<GstStreamFlags>(streamFlags | GST_STREAM_FLAG_SELECT));
-    }
 
     g_signal_connect_swapped(m_stream.get(), "notify::caps", G_CALLBACK(+[](VideoTrackPrivateGStreamer* track) {
         track->m_taskQueue.enqueueTask([track]() {
-            track->updateConfigurationFromCaps();
+            auto caps = adoptGRef(gst_stream_get_caps(track->m_stream.get()));
+            track->capsChanged(String::fromLatin1(gst_stream_get_stream_id(track->m_stream.get())), caps);
         });
     }), this);
     g_signal_connect_swapped(m_stream.get(), "notify::tags", G_CALLBACK(+[](VideoTrackPrivateGStreamer* track) {
+        auto tags = adoptGRef(gst_stream_get_tags(track->m_stream.get()));
         if (isMainThread())
-            track->updateConfigurationFromTags();
+            track->updateConfigurationFromTags(tags);
         else
-            track->m_taskQueue.enqueueTask([track]() {
-                track->updateConfigurationFromTags();
+            track->m_taskQueue.enqueueTask([track, tags = WTFMove(tags)]() {
+                track->updateConfigurationFromTags(tags);
             });
     }), this);
 
-    updateConfigurationFromCaps();
-    updateConfigurationFromTags();
+    auto caps = adoptGRef(gst_stream_get_caps(m_stream.get()));
+    updateConfigurationFromCaps(caps);
+
+    auto tags = adoptGRef(gst_stream_get_tags(m_stream.get()));
+    updateConfigurationFromTags(tags);
 }
 
-void VideoTrackPrivateGStreamer::updateConfigurationFromTags()
+void VideoTrackPrivateGStreamer::capsChanged(const String& streamId, const GRefPtr<GstCaps>& caps)
 {
     ASSERT(isMainThread());
-    if (!m_stream)
-        return;
+    updateConfigurationFromCaps(caps);
 
-    auto tags = adoptGRef(gst_stream_get_tags(m_stream.get()));
-    unsigned bitrate;
-    if (!tags || !gst_tag_list_get_uint(tags.get(), GST_TAG_BITRATE, &bitrate))
+    auto codec = m_player->codecForStreamId(streamId);
+    if (codec.isEmpty())
         return;
 
-    GST_DEBUG_OBJECT(m_stream.get(), "Setting bitrate to %u", bitrate);
     auto configuration = this->configuration();
-    configuration.bitrate = bitrate;
+    GST_DEBUG_OBJECT(objectForLogging(), "Setting codec to %s", codec.ascii().data());
+    configuration.codec = WTFMove(codec);
     setConfiguration(WTFMove(configuration));
 }
 
-void VideoTrackPrivateGStreamer::updateConfigurationFromCaps()
+void VideoTrackPrivateGStreamer::updateConfigurationFromTags(const GRefPtr<GstTagList>& tags)
 {
     ASSERT(isMainThread());
-    if (!m_stream)
+    GST_DEBUG_OBJECT(objectForLogging(), "Updating video configuration from %" GST_PTR_FORMAT, tags.get());
+    if (!tags)
         return;
 
-    auto caps = adoptGRef(gst_stream_get_caps(m_stream.get()));
-    if (!caps || !gst_caps_is_fixed(caps.get()))
-        return;
+    auto configuration = this->configuration();
+    auto scopeExit = makeScopeExit([&] {
```

Comment:
You don't need the scoped exit, do you?

Reply:
nop!

Reply:
I mean, not in `updateConfigurationFromTags()` but in `updateConfigurationFromCaps()` it is needed.